src/compat.rs

```diff
@@ -0,0 +1,83 @@
+use std::io::{Read, Write};
+use std::pin::Pin;
+use std::task::{Context, Poll};
+
+use tokio_io::{AsyncRead, AsyncWrite};
+
+#[derive(Debug)]
+pub(crate) struct AllowStd<S> {
+    pub(crate) inner: S,
+    pub(crate) context: *mut (),
```

Comment:
Why use void pointers and unsafe when you could use `Option<Context<'_>>`?

Maybe some comments explaining the reasoning behind all this would be good.

Reply:
This is taken from tokio-tls, so currently just trying to adapt rather than adjust since they're also adapting a read/write library there.

Reply:
Ok, I see. I also just realized that I confused with storing a Waker, which I usually do in order to wake up a task when some event happens. That is easier than storing `Context<'_>` because of the lifetime. Still think that ideally we avoid using unsafe for things like this.